Modules/_sqlite/connection.c

```diff
@@ -1998,10 +2003,21 @@ pysqlite_connection_iterdump_impl(pysqlite_Connection *self)
         }
         return NULL;
     }
+    PyObject *args[2] = {(PyObject *)self, filter};
+    PyObject *kwnames = Py_BuildValue("(s)", "filter");
+    if (!kwnames) {
+        goto error;
+    }
 
-    PyObject *retval = PyObject_CallOneArg(iterdump, (PyObject *)self);
+    PyObject *retval = PyObject_Vectorcall(iterdump, args, 1, kwnames);
```

Comment:
For the vectorcall, what you already wrote is perfectly fine, _but_ I'd prefer it if we followed the advice of PEP-590 and used `PyVectorcall_NARGS`; see other uses in connection.c for examples. This has implications for the `args` array and the values we pass to `PyObject_Vectorcall`.

Reply:
`PyVectorcall_NARGS` is not widely used, but I tried to add it.